### PR TITLE
Update variable_and_const.spec.ts shader to match current WGSL spec

### PR DESCRIPTION
**Pull request by zoddicus**

### CODE REVIEW

**Comment by zoddicus** [comment]
@kainino0x PTAL

File: src/webgpu/shader/validation/variable_and_const.spec.ts [hunk 1]
```diff
@@ -137,12 +137,37 @@ g.test('v_0038')
...
-      [[location(0)]] var<${storageClass}> a : ${type} = ${type}();
+    var code;
+    if (`${storageClass}` === 'in') {
+      code = `
+        struct MyInputs {
+          [[location(0)]] a : ${type};
+        };
+
+        [[stage(vertex)]]
+        fn main(inputs : MyInputs) {
```
Comment (jrprice) [inline src/webgpu/shader/validation/variable_and_const.spec.ts:148]: The WGSL spec now requires that all vertex shaders return at least `[[builtin(position)]]`.

File: src/webgpu/shader/validation/variable_and_const.spec.ts [hunk 2]
```diff
@@ -137,12 +137,37 @@ g.test('v_0038')
...
+        }
+      `;
+    } else if (`${storageClass}` === 'out') {
+      code = `
+        struct MyOutputs {
+          [[location(0)]] a : ${type};
```
Comment (jrprice) [inline src/webgpu/shader/validation/variable_and_const.spec.ts:154]: As above, this struct needs to include a position builtin.

File: src/webgpu/shader/validation/variable_and_const.spec.ts [hunk 3]
```diff
@@ -137,12 +137,37 @@ g.test('v_0038')
...
+        fn main() -> MyOutputs {
+          return MyOutputs();
         }
       `;
+    } else {
+      code = `
+      [[location(0)]] var<${storageClass}> a : ${type} = ${type}();
+
+      [[stage(vertex)]]
+      fn main() {
```
Comment (jrprice) [inline src/webgpu/shader/validation/variable_and_const.spec.ts:167]: As above, needs to return position builtin.

File: src/webgpu/shader/validation/variable_and_const.spec.ts (hunk 2 above)
Comment (jrprice) [reply src/webgpu/shader/validation/variable_and_const.spec.ts:154]: Actually, there's probably no reason this test needs to use a vertex shader, so could just make these all fragment shaders to keep things simpler.

File: src/webgpu/shader/validation/variable_and_const.spec.ts [hunk 4]
```diff
@@ -137,12 +137,37 @@ g.test('v_0038')
...
+
         [[stage(vertex)]]
-        fn main() {
+        fn main() -> MyOutputs {
+          return MyOutputs();
         }
       `;
+    } else {
+      code = `
+      [[location(0)]] var<${storageClass}> a : ${type} = ${type}();
```
Comment (jrprice) [inline src/webgpu/shader/validation/variable_and_const.spec.ts:164]: Presumably this is just for the `private` storage class case, in which case the `[[location(0)]]` decoration is not valid here.

File: src/webgpu/shader/validation/variable_and_const.spec.ts (hunk 1 above)
Comment (zoddicus) [reply src/webgpu/shader/validation/variable_and_const.spec.ts:148]: Converted to fragment

File: src/webgpu/shader/validation/variable_and_const.spec.ts (hunk 2 above)
Comment (zoddicus) [reply src/webgpu/shader/validation/variable_and_const.spec.ts:154]: Converted to fragment

File: src/webgpu/shader/validation/variable_and_const.spec.ts (hunk 4 above)
Comment (zoddicus) [reply src/webgpu/shader/validation/variable_and_const.spec.ts:164]: Done

File: src/webgpu/shader/validation/variable_and_const.spec.ts (hunk 3 above)
Comment (zoddicus) [reply src/webgpu/shader/validation/variable_and_const.spec.ts:167]: Converted to fragment

**Review by jrprice — APPROVED** [review]

**Review by kainino0x — APPROVED** [review]

**Comment by kainino0x** [comment]
Two tiny linting failures:
```
Error:   140:5  error  Unexpected var, use let or const instead  no-var
Error:   169:8  error  Insert `;`                                prettier/prettier
```
The former can be fixed manually and the latter with `npm run fix`. You can run the full CI locally with `npm test`.

**Pull request merged**